src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,205 @@
+        assertTrue(people.contains(user1));
+        assertTrue(people.contains(user2));
+        assertTrue(people.contains(user3));
+        assertEquals(3, people.size());
+    }
+
+    @Test
+    void return_CorrectUser_Ok() {
+        User returnedUser1 = registrationService.register(user1);
+        User returnedUser2 = registrationService.register(user2);
+        User returnedUser3 = registrationService.register(user3);
+        assertEquals(user1, returnedUser1);
+        assertEquals(user2, returnedUser2);
+        assertEquals(user3, returnedUser3);
+    }
+
+    @Test
+    void register_addUserIsNull_notOk() {
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(null);
+        });
+        assertEquals(0, people.size());
+    }
+
+    @Test
+    void register_addUserLoginIsNull_notOk() {
+        user1.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        assertEquals(0, people.size());
+    }
+
+    @Test
+    void register_addUserLoginStartWithNumber_notOk() {
+        user1.setLogin(STARTS_WITH_NUMBER);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        assertEquals(0, people.size());
+    }
+
+    @Test
+    void register_addUserLogInStartWithSymbol_notOK() {
+        user1.setLogin(STARTS_WITH_SYMBOL);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        assertEquals(0, people.size());
+    }
+
+    @Test
+    void register_addUserPasswordIsNull_notOk() {
+        user1.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        assertEquals(0, people.size());
+    }
+
+    @Test
+    void register_addUserPasswordIsShort_notOk() {
+        user1.setPassword(EMPTY_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(ONE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(TWO_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(THREE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(FOUR_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(FIVE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
```

Comment:
For negative scenario have one check it`s ok, you don`t need for every quantity of letters